contracts/prover/HyperProver.sol

```diff
@@ -26,36 +27,42 @@ contract HyperProver is IMessageRecipient, BaseProver, Semver {
      */
     event IntentAlreadyProven(bytes32 _intentHash);
 
+    /**
```

Comment:
The PROOF_TYPE string constant is less gas efficient than the previous enum approach. For on-chain comparison, this will cost more gas whenever the proof type is validated.